### PR TITLE
YLP-923 Clean-up go-common for v1 release

### PR DESCRIPTION
**Pull request by yvon-dblg**
- Change package name to github.com/mdblp/go-common
- Remove unused code
- Remove seagull GetPrivatePair(): Not implemented
- Remove client builder, and use a much simpler approach
- Add Dockerfile to ease local dev tests
- Add test coverage report
- Update README

### CODE REVIEW

File: clients/opa/opa.go [hunk 1]
```diff
@@ -12,26 +12,19 @@ import (
...
 	"strings"
 
-	"github.com/tidepool-org/go-common/clients/status"
+	"github.com/mdblp/go-common/clients/status"
 )
 
-// Client is the interface to opa.
-type Client interface {
+// API is the interface to opa.
+type API interface {
```
Comment (ydrisR) [inline clients/opa/opa.go]: Why changing the name of the type? It's called Client in all our services' clients...

**Review by ydrisR — COMMENTED** [review]
Globaly ok with the proposol but we should not change the name of the client interface to API. It's better to keep consistency. 
Also, how will these types and builders modification impact the other sevices (using these clients) ?

File: clients/seagull/seagull.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		return nil, errors.New("Missing SEAGULL_HOST environnement variable")
+	}
+	_, err := url.Parse(host)
+	if err != nil {
+		return nil, errors.New("Invalid SEAGULL_HOST url")
+	}
+	return &Client{
+		host:       host,
+		httpClient: httpClient,
+	}, nil
```
Comment (CedricPerotto) [inline clients/seagull/seagull.go]:
```suggestion
	return NewClient(httpClient, host)
```

File: Dockerfile.test [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# To retreive the coverage report run:
+# docker run go-common:test > coverage.html
+FROM mongo:4.2
+RUN apt-get update -y && \
+    apt-get install -y --no-install-recommends apt-utils && \
+    apt-get upgrade -y && \
+    apt-get install -y --no-install-recommends wget build-essential psmisc && \
+    apt-get clean -y
+WORKDIR /tmp
+ENV GO_ARCHIVE go1.16.6.linux-amd64.tar.gz
```
Comment (CedricPerotto) [inline Dockerfile.test]: we use GO 1.16 in tests but GO 1.15 in the module... Just keep one version

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 Go-Common is the library of common functions for Tidepool's Go-based applications
 
+## Unreleased
```
Comment (CedricPerotto) [inline CHANGELOG.md]: Let's bump a release

**Review by CedricPerotto — CHANGES_REQUESTED** [review]

File: Dockerfile.test (hunk 3 above)
Comment (yvon-dblg) [reply Dockerfile.test]: This file is just for local testing, not use in Jenkins.
To see if it keeps working on the next version

**Review by CedricPerotto — APPROVED** [review]

**Pull request merged**